### PR TITLE
ability to provide alternative extract text plugin

### PR DESCRIPTION
**Pull request by VyvIT**
This change is [Reviewable](https://reviewable.io/reviews/shakacode/bootstrap-loader/340)

### CODE REVIEW

**Comment by justin808** [comment]
@VyvIT this looks intriguing. In order to merge, I'd need:

1. A way that I can verify this. Can you place some examples?
2. See https://github.com/shakacode/bootstrap-loader/blob/master/CONTRIBUTING.md
3. I'd need some example, readme and changlog entries.

**Comment by neodescis** [comment]
Should the default be mini-css-extract-plugin instead of extract-text-webpack-plugin? And also swap the dependency?

**Comment by VyvIT** [comment]
Updated the docs and created an example based of of `basic` example, but was unable to run the build task as I'm getting 'unable to find 'boostrap-sass' module error (maybe because I'm on Win OS).
@justin808 Let me know if that works for you.

**Comment by VyvIT** [comment]
@neodescis That's up to bootstrap-loader maintainer(s) decision. It would be a breaking change. But I agree with you.

**Comment by neodescis** [comment]
Certainly it would be a breaking change. However, this dependency on the now-deprecated extract-text-webpack-plugin is currently the only thing preventing me from migrating a project to Webpack 4. It seems to me that it's really the only path forward for this project if it is to remain relevant to the community.

**Comment by justin808** [comment]
@neodescis @VyvIT breaking is fine. Just requires a major version bump.

@alexfedoseev?

**Comment by justin808** [comment]
@VyvIT I need you to troubleshoot:

> Updated the docs and created an example based of of basic example, but was unable to run the build task as I'm getting 'unable to find 'boostrap-sass' module error (maybe because I'm on Win OS).

I'm happy to merge once I can install the 3 example directories and run a set of steps to try out both the commands in each of 3 README.md files.

@neodescis any chance you can give us a hand?

**Comment by alexfedoseev** [comment]
`mini-css-extract-plugin` should be default, yeah.

File: examples/basic-with-mini-css-plugin/.babelrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": ["es2015"]
```
Comment (yvan33) [inline examples/basic-with-mini-css-plugin/.babelrc]: Shouldn't it we use `env` instead of `es2015`. `es2015` is deprecated

**Comment by neodescis** [comment]
@justin808 Happy to help, but I'm out at a conference this week.

File: examples/basic-with-mini-css-plugin/.babelrc (hunk 1 above)
Comment (graingert) [reply examples/basic-with-mini-css-plugin/.babelrc]: yes you should use env

**Comment by justin808** [comment]
@VyvIT Any update? Can anybody run the examples?

**Comment by neodescis** [comment]
@justin808 I can take a look this coming week, but I think the PR needs to be updated to swap out the default extract plugin. Either @VyvIT can do this, or I can create my own PR.

**Comment by neodescis** [comment]
@justin808 Hmm, well I had a few minutes to look at it tonight, and I see what @VyvIT means about the examples not being Windows-friendly. Specifically, the npm scripts don't run. At any rate, it's looking to me like the project needs to update a bunch more dependencies than just the extract plugin! Unfortunately I don't think I'm going to have time to bite off that much work.

**Comment by alexfedoseev** [comment]
Thanks for PR but I'm going to close it in favor of https://github.com/shakacode/bootstrap-loader/pull/347

**Pull request closed without merging**